crates/error/src/compile.rs

```diff
@@ -25,6 +27,9 @@ pub enum Error {
     #[error("Could not create Rust string from C string: {_0}")]
     CStrConversionError(#[from] Utf8Error),
 
+    #[error("Could not query the Salsa database: {0:?}")]
+    SalsaDbError(DiagnosticAdded),
+
```

Comment:
Please don't add new variants to this error enum. This enum is for errors in the _LLVM to FLO_ compiler. Same for below.

You can rename this module to `src/llvm_compile.rs` and then please create a new one `src/cairo_compile.rs` that provides the same infrastructure and integration into the top level `Error` type as the other ones. Your crate should _only_ depend on the new `hieratika_errors::cairo_compile::{Error, Result}` types.

Reply:
Alternatively you can have `hieratika_errors::compile::cairo` and `hieratica_errors::compile::llvm` as the two error modules here.